File: LogicOverSyntaxCodes/Searching/PythonCodes/Leetcode540.py
class Solution:
    def singleNonDuplicate(self, nums):
        start, end = 0, len(nums) - 1

        while start < end:
            mid = start + (end - start) // 2
            which_index = mid + 1 if mid % 2 == 0 else mid - 1
            
            if nums[mid] == nums[which_index]:
                start = mid + 1
            else:
                end = mid

        return nums[start]

File: Searching/JavaCodes/Leetcode540.java
class Solution {
   public static int singleNonDuplicate(int[] nums) {
        int start=0,end= nums.length-1;

        while(start<end){
            int mid=start+(end-start)/2;
            int whichIndex=mid%2==0?mid+1:mid-1;
            if(nums[mid]==nums[whichIndex]){
                start=mid+1;
            }
            else{
                end=mid;
            }
        }
        return nums[start];
    }
}